lib/index_group.rb

```diff
@@ -152,7 +166,48 @@ def name_pattern
 
     def delete(index_name)
       logger.info "Deleting index #{index_name}"
-      @client.indices.delete(index: index_name)
+      @long_timeout_client.indices.delete(index: index_name)
+    end
+
+    # Sort the index map by date, remove the most recent two (active and latest inactive) and return hash
+    def cleaned_aliases
+      value = alias_map(include_closed: true).sort
+      value.pop(2) # Fast logic! Currently will always have one active + one closed index!
+      value.to_h
+    end
+
+    def find_last_update(name)
+      last_update_raw = @client.search(index: name, body: last_update_query)
+      last_update_hit = last_update_raw.dig("hits", "hits")
+      last_update_hit[0]&.dig("_source", "updated_at")
+    end
+
+    # Finds the last time an index was updated based on the last updated_at time of the last edited
+    # item in the index. This shows last time it was open for use.
+    def last_update_query
+      {
+        "_source" => "updated_at",
+        "size" => 1,
+        "sort" => [
+          {
+            "updated_at" =>
+            {
+              "order" => "desc",
+              "unmapped_type" => "date",
+            },
+          },
+        ],
+      }
+    end
+
+    def delete_based_on_age_check(name, day_limit, last_update)
+      age = Time.now.to_i - last_update.to_datetime.to_f # Everything is handled in seconds, thanks Ruby!
+      days = age.to_i / 86_400 # One day in seconds
+
+      return unless days >= day_limit.to_i
+
+      logger.info "Auto-cleaning index #{name} which is unused and #{days} days old"
+      delete(name)
```

Comment:
The other `clean` method has a guard clause on the delete:
```
delete(name) if details.fetch("aliases", {}).empty?
```
Is this something we want to consider?

Reply:
Done